Extensions/Optimizely/Src/Tasks/OptimizelyX/operations/TaskOperation.ts

```diff
@@ -0,0 +1,196 @@
+      let projectId: string = this.taskInputs.getProjectId();
+
+      if(this.taskInputs.isWebProject()) {
+          experiment.status = OptimizelyExperimentStatus.Running;
+      } else {
+          let environment: IOptimizelyEnvironment = await this.optimizelyClient.getEnvironment(projectId, environmentName);
+          if (environment == null) {
+              throw tl.loc("EnvironmentNotFound", environmentName);
+          }
+
+          Object.keys(experiment.environments).forEach((value) => {
+              if (value.toLowerCase() == environmentName.toLowerCase()) {
+                  experiment.environments[value].status = OptimizelyExperimentStatus.Running;
+              }
+          });
+      }
+
+      let experimentName = experiment.name;
+      await this.optimizelyClient.updateExperiment(experiment.id, experiment);
+
+      let counter: number = 6;
+      while (counter > 0) {
+          console.log(tl.loc("WaitingForExperimentToStart", experimentName));
+          await this.sleep(10000);
+          counter--;
+      }
+
+      console.log(tl.loc("ABTestStartedSuccessfully", experimentName));
+      console.log(tl.loc("VisitOptimizelyPage") + ` https://app.optimizely.com/v2/projects/${projectId}/experiments`);
+  }
+
+  public async stopABTest(experiment: IOptimizelyAlphaBetaTest): Promise<void> {
+      let environmentName: string = this.taskInputs.getEnvironmentName();
+      let projectId: string = this.taskInputs.getProjectId();
+
+      if(this.taskInputs.isWebProject()) {
+          experiment.status = OptimizelyExperimentStatus.Paused;
+      } else {
+          let environment: IOptimizelyEnvironment = await this.optimizelyClient.getEnvironment(projectId, environmentName);
+          if (environment == null) {
+              throw tl.loc("EnvironmentNotFound", environmentName);
+          }
+
+          Object.keys(experiment.environments).forEach((value) => {
+              if (value.toLowerCase() == environmentName.toLowerCase()) {
+                  experiment.environments[value].status = OptimizelyExperimentStatus.Paused;
+              }
+          });
+      }
+
+      let experimentName = experiment.name;
+
+      await this.optimizelyClient.updateExperiment(experiment.id, experiment);
+      console.log(tl.loc("ExperimentSuccessfullyPaused", experimentName));
+  }
+
+  public updateTrafficVariation(experiment: IOptimizelyAlphaBetaTest): IOptimizelyAlphaBetaTest {
+      let totalTrafficParameter : string = this.taskInputs.getTotalTraffic();
+      let totalTraffic: number = Number.parseInt(totalTrafficParameter);
+
+      if (Number.isNaN(totalTraffic)) {
+          throw tl.loc("TrafficValueNotValid", totalTrafficParameter);
+      }
+
+      if (totalTraffic > 100) {
+          throw tl.loc("TrafficValueCantExceed", totalTraffic);
+      }
+
+      let holdBack: number = 10000 - (totalTraffic * 100);
+      experiment.holdback = holdBack;
```

Comment:
What is this holdback?

Reply:
This is the percentage of overall traffic from whom you want to withhold the rollout of the experiment. i.e. if the holdback % is 40, this means that only 60% of the total traffic will be eligible for the experiment.